Create rule S6293: Using a biometric authentication independent of a cryptographic solution is security-sensitive (#92)
SonarSource/rspec: rules/S6293/description.adoc

> Android comes with Android KeyStore, a secure container for storing key materials. It's possible to define certain keys to be unlocked only by authenticated users, with biometric credentials. In this way, even if the application process is compromised, the attacker cannot access keys, the presence of the authorized user is required.

Reviewer: ```suggestion
Android comes with Android KeyStore, a secure container for storing key materials. It's possible to define certain keys to be unlocked only when users authenticate using biometric credentials. This way, even if the application process is compromised, the attacker cannot access keys, as the presence of the authorized user is required.
```

> These keys can be used, to encrypt, sign or create a message authentication code (MAC) as proof that the authentication result has not been tampered with, ie: an attacker is not able to directly call the ``++onAuthenticationSucceeded++`` method, used normally as a callback after successful authentications, or the underlying critical operations implemented by this callback.

Reviewer: ```suggestion
These keys can be used, to encrypt, sign or create a message authentication code (MAC) as proof that the authentication result has not been tampered with. This protection defeat the scenario where an attacker with physical access to the device would try to hook into the application process and to call the ``++onAuthenticationSucceeded++`` method directly. Therefore he would be unable to extract the sensitive data or to perform the critical operations protected by the biometric authentication.
```

== Ask Yourself Whether

> * Cryptographic keys need to be protected using biometric authentication.
> * Sensitive information is required to be accessed by authenticated users, with biometric credentials only.

Reviewer: ```suggestion
The application contains:
* Cryptographic keys that need to be protected using biometric authentication.
* Sensitive information that are require biometric authentication to be accessed.
```
